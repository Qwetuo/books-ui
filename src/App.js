// import React, { Component } from 'react';
// import logo from './logo.svg';
// import './App.css';

// class App extends Component {
//   render() {
//     return (
//       <div className="App">
//         <header className="App-header">
//           <img src={logo} className="App-logo" alt="logo" />
//           <h1 className="App-title">Welcome to React</h1>
//         </header>
//         <p className="App-intro">
//           To get started, edit <code>src/App.js</code> and save to reload.
//         </p>
//       </div>
//     );
//   }
// }

// export default App;

import React, { Component } from "react";
import "./App.css";

class App extends Component {
  constructor() {
    super();
    this.state = {
      books: []
    };
  }
  render() {
    return (
      <div>
        {this.state.books.map(book => {
          return <li key={book._id}>{book.title}</li>;
        })}
      </div>
    );
  }
  async componentDidMount() {
    const response = await fetch(
      process.env.REACT_APP_BOOKS_API || "http://localhost:3000/books"
    );
    console.log("response", response);
    const books = await response.json();
    console.log("books", books);
    this.setState({
      books: books
    });
  }
}

export default App;
