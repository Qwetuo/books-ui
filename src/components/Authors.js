import React, { Component } from "react";

const API_HOST = process.env.REACT_APP_BOOKS_API || "http://localhost:3000";

class Authors extends Component {
  constructor() {
    super();
    this.state = {
      authors: []
    };
  }

  async getAuthors() {
    const url = `${API_HOST}/authors`;
    const response = await fetch(url);
    if (response.ok) {
      const data = await response.json();
      this.setState({
        authors: data
      });
    }
  }

  async componentDidMount() {
    await this.getAuthors();
  }

  render() {
    return (
      <div>
        <h1>AUTHORS</h1>
        {this.state.authors.map(author => {
          return <li key={author._id}>{author.name}</li>;
        })}
      </div>
    );
  }
}

export default Authors;
