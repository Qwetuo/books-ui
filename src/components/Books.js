import React, { Component } from "react";

const API_HOST = process.env.REACT_APP_BOOKS_API || "http://localhost:3000";

class Books extends Component {
  constructor() {
    super();
    this.state = {
      books: []
    };
  }

  async getBooks() {
    const url = `${API_HOST}/books`;
    const response = await fetch(url);
    if (response.ok) {
      const data = await response.json();
      this.setState({
        books: data
      });
    }
  }

  async componentDidMount() {
    await this.getBooks();
  }

  render() {
    return (
      <div>
        <h1>BOOKS</h1>
        {this.state.books.map(book => {
          return <li key={book._id}>{book.title}</li>;
        })}
      </div>
    );
  }
}

export default Books;
