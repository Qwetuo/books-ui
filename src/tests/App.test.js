import React from 'react';
import ReactDOM from 'react-dom';
import App from '../App';
import Authors from '../components/Authors';
import Books from '../components/Books';
import ShallowRenderer from 'react-test-renderer/shallow'; // ES6


it.skip('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders App', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<App />);
  const result = renderer.getRenderOutput();
  expect(result).toMatchSnapshot()
})

it('renders Authors', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<Authors />);
  const result = renderer.getRenderOutput();
  expect(result).toMatchSnapshot()
})

it('renders Books', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<Books />);
  const result = renderer.getRenderOutput();
  expect(result).toMatchSnapshot()
})

