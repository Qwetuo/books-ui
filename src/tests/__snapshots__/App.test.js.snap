// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App 1`] = `
<div>
  <Books />
  <Authors />
</div>
`;

exports[`renders Authors 1`] = `
<div>
  <h1>
    AUTHORS
  </h1>
</div>
`;

exports[`renders Books 1`] = `
<div>
  <h1>
    BOOKS
  </h1>
</div>
`;
